src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -773,14 +787,13 @@ void IODataStreamListener.StreamRequested()
         {
         }
 
-
         /// <summary>
         /// This method is called when an async stream is requested. It is a no-op.
         /// </summary>
         /// <returns>A task for method called when a stream is requested.</returns>
         Task IODataStreamListener.StreamRequestedAsync()
         {
-            return TaskUtils.GetTaskForSynchronousOperation(() => ((IODataStreamListener)this).StreamRequested());
+            return TaskUtils.CompletedTask;

```

Comment:
You can remove the `IODataStreamListener.StreamRequested()` if it's not needed anymore

Reply:
It'd not be safe to remove the method since this class implements `IODataOutputInStreamErrorListener` hence the requirement to implement all the methods including `StreamRequested` and `StreamRequestedAsync`. Its actually an extensibility hook, just like `StreamDisposed` and `StreamDisposedAsync`